include/openthread/platform/radio.h

```diff
@@ -737,6 +737,28 @@ uint64_t otPlatRadioGetNow(otInstance *aInstance);
  */
 uint32_t otPlatRadioGetBusSpeed(otInstance *aInstance);
 
+/**
+ * Get the bus latency in microseconds between the host and the radio chip.
+ *
+ * @param[in]   aInstance    A pointer to an OpenThread instance.
+ *
+ * @returns The bus latency in microseconds between the host and the radio chip.
+ *          Return 0 when the MAC and above layer and Radio layer resides on the same chip.
+ *
+ */
+uint32_t otPlatRadioGetBusLatency(otInstance *aInstance);
+
+/**
+ * Set the bus latency in microseconds between the host and the radio chip.
+ *
+ * @note This function does not do anything when the MAC and above layer and Radio layer resides on the same chip.
+ *
+ * @param[in]   aInstance    A pointer to an OpenThread instance.
+ * @param[in]   aLatency     Bus latency in microseconds.
+ *
+ */
+void otPlatRadioSetBusLatency(otInstance *aInstance, uint32_t aLatency);
```

Comment:
Thanks @maciejbaczmanski.

Do we need a `SetBusLatency` API? Should the OpenThread stack have the ability to modify this parameter on the platform?

In my opinion, "bus latency" is a platform-specific parameter. The OpenThread stack should be able to retrieve this value (to account for it in CSL transmissions) but not modify it directly.

Another consideration is whether we expect the "bus latency" to change dynamically during runtime. If so, we could potentially add a callback mechanism for the platform to notify the OpenThread stack of any bus latency changes.

What are your thoughts on this?

Reply:
Thanks for input @abtink!

So basically this approach was a result of our discussion with @Irving-cl in an issue created earlier, and we came up with a conclusion that it could be managed only on host's side https://github.com/openthread/openthread/issues/10410#issuecomment-2196649268
API and CLI command for setting the value were meant as a helper for fine tuning the mechanism


> In my opinion, "bus latency" is a platform-specific parameter. The OpenThread stack should be able to retrieve this value (to account for it in CSL transmissions) but not modify it directly.
> 
> Another consideration is whether we expect the "bus latency" to change dynamically during runtime. If so, we could potentially add a callback mechanism for the platform to notify the OpenThread stack of any bus latency changes.

Do you think that we should receive the latency from the RCP device then? 
If so, I've proposed similar approach here (but didn't come up with a callback, which sound good to me!): https://github.com/openthread/openthread/issues/10410#issuecomment-2196606308  @Irving-cl what are your thoughts?

Reply:
> Do you think that we should receive the latency from the RCP device then?

Not necessarily. The OpenThread stack should obtain this value from the radio platform layer. The platform layer implementation can then determine whether it's tracked on the host or the RCP. Practically speaking, it makes sense to me for it to be tracked on the host-side radio platform (e.g., as part of the `RadioSpinel` module).

> API and CLI command for setting the value were meant as a helper for fine tuning the mechanism

I understand the benefit of a CLI command for fine-tuning during integration/testing, but adding it as a general-purpose command might be confusing. This isn't a parameter that typical users of a Thread device should be able to modify during operation.

Instead, I propose adding support for setting the bus latency as a `diag` CLI sub-command, implemented in the `RadioSpinel::PlatDiagProcess()` function on the platform layer. This would provide a convenient way to fine-tune the value during testing without exposing it in the main API/CLI. 

Thoughts?

Reply:
Oh, didn't think about PlatDiagProcess, that makes sense! thanks

Reply:
> Instead, I propose adding support for setting the bus latency as a `diag` CLI sub-command, implemented in the `RadioSpinel::PlatDiagProcess()` function on the platform layer. This would provide a convenient way to fine-tune the value during testing without exposing it in the main API/CLI.

I've checked the code and I'm wondering if handling the subcommand in `otPlatDiagProcess()` in `posix/radio.cpp` wouldn't be a better place?
- `RadioSpinel::PlatDiagProcess()` receives a String command (packed in `otPlatDiagProcess()`), which we would need to parse back to args, so that adds operations that cancel each other out.
- `RadioSpinel::SetBusLatency()` will be called from `Posix::Radio::Init`  as per @Irving-cl 's suggestion, so the method itself needs can't be set as private in `RadioSpinel` class.

`otPlatRadioSetBusLatency` and "non-diag CLI" commands will be obviously removed, so I guess we won't directly expose the functionality?

Reply:
> I've checked the code and I'm wondering if handling the subcommand in `otPlatDiagProcess()` in `posix/platform/radio.cpp` wouldn't be a better place?

This sounds good to me and it can work.

Another approach could be to add a new `PlatDiagProcess()` overload that takes the full `args` array and handles combining them into a string, instead of doing this within `otPlatDiagProcess()`. This would allow us to add more custom `RadioSpinel`-specific diag sub-commands in the future.

Reply:
@abtink thanks, I've decided to proceed with your suggestion to add `RadioSpinel` specific commands and add `PlatDiagProcess()` overload.